lib/mongo/cluster.rb

```diff
@@ -194,10 +198,10 @@ def initialize(seeds, monitoring, options = Options::Redacted.new)
     # @return [ Proc ] The Finalizer.
     #
     # @since 2.2.0
-    def self.finalize(pools, cursor_reaper)
+    def self.finalize(pools, periodic_executor)
       proc do
-        begin; cursor_reaper.kill_cursors; rescue; end
-        cursor_reaper.stop!
+        periodic_executor.stop!
+        begin; periodic_executor.execute; rescue; end
```

Comment:
It looks like these two lines will call CursorReaper#kill_cursors twice, once through the `finalize` alias in the call to PeriodicExecutor#stop! and then again through the call to the `execute` alias through PeriodicExecutor#execute. It doesn't look like calling it twice causes any issues, although since it does require acquiring a lock on a mutex, I'm wondering if it might have some performance cost.